src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint.py

```diff
@@ -330,6 +333,51 @@ def finalize_run(node_states: Dict[str, NodeStatus]) -> None:
             # as the pipeline run status will already have been published.
             pass
 
+    def check_pipeline_cancellation() -> bool:
+        """Check if the pipeline should continue execution.
+
+        Returns:
+            True if execution should continue, False if it should stop.
+        """
+        try:
+            # Fetch the current pipeline run status
+            list_args: Dict[str, Any] = {}
+            if args.run_id:
+                list_args = dict(id=UUID(args.run_id))
+            else:
+                list_args = dict(orchestrator_run_id=orchestrator_pod_name)
+
+            pipeline_runs = client.list_pipeline_runs(
+                hydrate=False,  # We only need status, not full hydration
+                project=deployment.project_id,
+                deployment_id=deployment.id,
+                **list_args,
+            )
+            if not len(pipeline_runs):
+                # No pipeline run found, assume we should continue
+                return True
+
+            pipeline_run = pipeline_runs[0]
```

Comment:
Also with my latest changes that are merged in here, you are guaranteed to have a `pipeline_run` already (defined somewhere above). I think in this function, you should just re-fetch this pipeline run, no need to run any `list_pipeline_runs(...)` calls